tests/src/com/unciv/logic/map/UnitMovementAlgorithmsTests.kt

```diff
@@ -235,7 +236,7 @@ class UnitMovementAlgorithmsTests {
         city.civInfo = otherCiv
         tile.owningCity = city
 
-        unit.baseUnit = BaseUnit().apply { unitType = UnitType.Melee }
+        unit.baseUnit = BaseUnit().apply { unitType = ruleSet.unitTypes.keys.random(); ruleset = ruleSet }
```

Comment:
Unpredictability in a unit test? Since that's in `canNOTPassForeignTiles` I'd make that a `first()`, since all alternatives would serve equally well for the tests's goal